### ISSUE TITLE
Which model type should I use for financial price prediction?

### ISSUE THREAD

**Issue by ErcinDedeoglu** [dotnet/machinelearning#498]
First of all thank you for the great library!
My question is simple: I want to predict next period price with pre-computed history values.
I have over 30 rows data for each price.
Price and datas are decimal.

For example history:
Indicator1 - Indicator 2 - Indicator 3 - Price - **Trend**
10,01121 - 23,56540 - 12.00001 - 12,23321 - UP
9,00001 - 3,00040 - 2.00001 - 1,23300 - DOWN
...
...
And data to predict coming like
8,11211 - 1,00020 - 0.00021 - 3,5555 - ?
I want to get **TREND** field.

Which model should I use? Any example will be perfect?
Regards!

**Comment by justinormont** [comment]
You turned this into a binary classification problem (UP vs. DOWN), so any of the binary classifiers should give you a working model. 

I'm unsure which are exposed w/ a nice ML.NET interface, but the repo has code for these:

-   AveragedPerceptron: Averaged Perceptron
-   BinaryClassificationGamTrainer: Generalized Additive Model for Binary Classification
-   BinarySGD: Hogwild SGD (binary)
-   FastForestClassification: Fast Forest Classification
-   FastTreeBinaryClassification: FastTree (Boosted Trees) Classification
-   FieldAwareFactorizationMachine: Field-aware Factorization Machine
-   LightGBMBinary: LightGBM Binary Classifier
-   LinearSVM: SVM (Pegasos-Linear)
-   LogisticRegression: Logistic Regression
-   PriorPredictor: Prior Predictor
-   RandomPredictor: Random Predictor
-   SDCA: Fast Linear (SA-SDCA)
-   WeightedEnsemble: Parallel Ensemble (bagging, stacking, etc)

Also, of notable importance, you'll want to ensure that your Train & Test datasets are split on time (all newer rows are in the test set). Splitting on company (train & test dataset don't have any companies in common) is almost as good, but will still leak some information like raising tide effects.

**Comment by abgoswam** [comment]
DRI RESPONSE : Looks like question has been answered. Plan to close this issue in the next 1-2 days
